ibm-messaging-mq-cloud-showcase-app/backend/AWS-showcase-backend/models/Responder.js

```diff
@@ -34,12 +34,27 @@ class Responder {
             this.mqclient.get(queueName,1,null, 'DYNAMIC', this.appId)
             .then((messages) => {
                 debug_warn(`The producer retrieved this message ${JSON.stringify(messages)}`);
-                if(!messages[0].replyToMsg) {
+                if (messages.length === 0) {
+                    debug_info("Poison Message");
+                    resolve();
+                } else if (!messages[0].replyToMsg) {
                     debug_warn("This is not a reply to queue valid message");
                     resolve(null);
+                } else if (messages[0].replyToMsg) {
+                    this.mqclient.checkQueueExists(messages[0].replyToMsg, this.appId)
+                    .then((flag) => {
+                        if (flag === false) {
+                            debug_info("Some Poison Message");
+                        } else {
+                            debug_info(`Responder ${this.myID} obtained message from queue`);
+                        }
+                        resolve(messages);
```

Comment:
If we are resolving the same way regardless of whether the reply to queue exists, what is the impact of invoking `checkQueueExists` ?